src/net_processing.cpp

```diff
@@ -2383,6 +2383,24 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 }
             }
         }
+        // If we're in IBD, we want outbound peers that will serve us a useful
+        // chain. Disconnect peers that are on chains with insufficient work.
+        if (IsInitialBlockDownload() && nCount != MAX_HEADERS_RESULTS) {
+            // When nCount < MAX_HEADERS_RESULTS, we know we have no more
+            // headers to fetch from this peer. Compare their tip to
+            // nMinimumChainWork (rather than chainActive.Tip()) because we
+            // won't start block download until we have a headers chain that
+            // has at least nMinimumChainWork, even if a peer has a chain past
+            // our tip, as an anti-DoS measure.
+            if (nodestate->pindexBestKnownBlock->nChainWork < nMinimumChainWork) {
```

Comment:
I'm unsure if it would be possible to get here without setting a pindexBestKnownBlock, but please double-check to be safe: ```if (nodestate->pindexBestKnownBlock && ...```

Reply:
Done